src/optimized_transaction.rs

```diff
@@ -336,29 +339,55 @@ impl Helius {
     pub async fn send_smart_transaction(&self, config: SmartTransactionConfig<'_>) -> Result<Signature> {
         let (transaction, last_valid_block_height) = self.create_smart_transaction(&config.create_config).await?;
 
-        // Common logic for sending transactions
-        let send_transaction_config = config.send_options.clone();
-
-        let send_result = |transaction: &Transaction| {
-            self.connection()
-                .send_transaction_with_config(transaction, send_transaction_config)
-        };
-        let send_versioned_result = |transaction: &VersionedTransaction| {
-            self.connection()
-                .send_transaction_with_config(transaction, send_transaction_config)
-        };
+        match transaction {
+            SmartTransaction::Legacy(tx) => {
+                self.send_and_confirm_transaction(
+                    &tx,
+                    config.send_options,
+                    last_valid_block_height,
+                    Some(config.timeout.into()),
+                )
+                .await
+            }
+            SmartTransaction::Versioned(tx) => {
+                self.send_and_confirm_transaction(
+                    &tx,
+                    config.send_options,
+                    last_valid_block_height,
+                    Some(config.timeout.into()),
+                )
+                .await
+            }
+        }
+    }
 
+    /// Sends a transaction and handles its confirmation status
+    ///
+    /// # Arguments
+    /// * `transaction` - The transaction to be sent, which implements `SerializableTransaction`
+    /// * `send_transaction_config` - Configuration options for sending the transaction
+    /// * `last_valid_block_height` - The last block height at which the transaction is valid
+    /// * `timeout` - Optional duration for polling transaction confirmation, defaults to 60 seconds
+    ///
+    /// # Returns
+    /// The transaction signature, if successful
+    pub async fn send_and_confirm_transaction(
+        &self,
+        transaction: &impl SerializableTransaction,
+        send_transaction_config: RpcSendTransactionConfig,
+        last_valid_block_height: u64,
+        timeout: Option<Duration>,
```

Comment:
what do we think about this Option<Duration> timeout argument? I saw you pushed a commit but it failed in ci